G1ANT.Addon.Net/SimplifiedMessageSummary.cs

```diff
@@ -1,27 +1,321 @@
+using G1ANT.Addon.Net.Models;
 
 namespace G1ANT.Addon.Net
 {
-    public class SimplifiedMessageSummary : MessageSummary
+    public class SimplifiedMessageSummary
     {
-        public new string TextBody;
         public IMailFolder Folder;
-        public ListStructure MessageAttachments;
 
-        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder, ListStructure attachmentsList) : base(summary.Index)
+        private MessageSummary messageSummary = null;
+        private MimeMessage fullMessage = null;
+
+        public SimplifiedMessageSummary()
         {
-            TextBody = ParseBody(summary, folder);
-            MessageAttachments = attachmentsList;
-            Envelope = summary.Envelope;
-            UniqueId = summary.UniqueId;
+            FullMessage = new MimeMessage();
+        }
+
+        public SimplifiedMessageSummary(MessageSummary summary, IMailFolder folder)
+        {
+            messageSummary = summary;
             Folder = folder;
         }
 
-        private static string ParseBody(IMessageSummary message, IMailFolder folder)
+        protected SimplifiedMessageSummary(MimeMessage message)
+        {
+            FullMessage = message;
+        }
+
+        public MimeMessage FullMessage
+        {
+            get
+            {
+                if (fullMessage == null && messageSummary != null)
+                    fullMessage = Folder.GetMessage(messageSummary.UniqueId);
+                return fullMessage;
+            }
+            set
+            {
+                fullMessage = value;
+            }
+        }
+
+        public List<object> Attachments
+        {
+            get => CreateAttachmentStructuresFromAttachments(FullMessage.Attachments);
+            set
+            {
+                var bodyBuilder = new BodyBuilder();
+                if (bodyBuilder.Attachments != null)
+                {
+                    bodyBuilder.HtmlBody = HtmlBody;
+                    bodyBuilder.TextBody = TextBody;
+                    foreach (var attachment in value)
+                        bodyBuilder.Attachments.Add(attachment.ToString());
+                    FullMessage.Body = bodyBuilder.ToMessageBody();
+                }
+            }
+        }
+
+        private List<object> CreateAttachmentStructuresFromAttachments(IEnumerable<MimeEntity> attachments)
+        {
+            List<object> attachmentsList = new List<object>();
+
+            foreach (var attachment in attachments.Where(x => !string.IsNullOrEmpty(x.ContentDisposition?.FileName)))
+            {
+                AttachmentModel attachmentModel = new AttachmentModel(attachment);
+                AttachmentStructure temp = new AttachmentStructure(attachmentModel);
+                attachmentsList.Add(temp);
+            }
+            return attachmentsList;
+        }
+
+        public string MessageId
+        {
+            get
+            {
+                if (fullMessage != null)
```

Comment:
There's something wrong with multiple null checks in every property. It looks like you try to make this model a container for two different objects. Consider adding an interface or abstract model and use it in two models that are containers just for one object (one for `MessageSummary`, one for `MimeMessage`). Or event don't bother with interface.
Otherwise you end up with lots of checks in lots of methods.

Reply:
I was thinking about that but...SimplifiedMessageSummary would have to wrap this interface and pass calls to it, depending on level of data needed it should then manage switching between different implementations of the interface.
It's too much hassle